pl_bolts/models/rl/ppo_model.py

```diff
@@ -0,0 +1,378 @@
+            act_dim = self.env.action_space.shape[0]
+            actor_mlp = MLP(self.env.observation_space.shape, act_dim)
+            self.actor = ActorContinous(actor_mlp, act_dim)
+        elif isinstance(self.env.action_space, gym.spaces.discrete.Discrete):
+            actor_mlp = MLP(self.env.observation_space.shape, self.env.action_space.n)
+            self.actor = ActorCategorical(actor_mlp)
+        else:
+            raise NotImplementedError(
+                'Env action space should be of type Box (continous) or Discrete (categorical). '
+                f'Got type: {type(self.env.action_space)}'
+            )
+
+        self.batch_states = []
+        self.batch_actions = []
+        self.batch_adv = []
+        self.batch_qvals = []
+        self.batch_logp = []
+
+        self.ep_rewards = []
+        self.ep_values = []
+        self.epoch_rewards = []
+
+        self.episode_step = 0
+        self.avg_ep_reward = 0
+        self.avg_ep_len = 0
+        self.avg_reward = 0
+
+        self.state = torch.FloatTensor(self.env.reset())
+
+    def forward(self, x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """
+        Passes in a state x through the network and returns the policy and a sampled action
+
+        Args:
+            x: environment state
+
+        Returns:
+            Tuple of policy and action
+        """
+        pi, action = self.actor(x)
+        value = self.critic(x)
+
+        return pi, action, value
+
+    def discount_rewards(self, rewards: List[float], discount: float) -> List[float]:
+        """Calculate the discounted rewards of all rewards in list
+
+        Args:
+            rewards: list of rewards/advantages
+            discount: discount factor
+
+        Returns:
+            list of discounted rewards/advantages
+        """
+        assert isinstance(rewards[0], float)
+
+        cumul_reward = []
+        sum_r = 0.0
+
+        for r in reversed(rewards):
+            sum_r = (sum_r * discount) + r
+            cumul_reward.append(sum_r)
+
+        return list(reversed(cumul_reward))
+
+    def calc_advantage(self, rewards: List[float], values: List[float], last_value: float) -> List[float]:
+        """Calculate the advantage given rewards, state values, and the last value of episode
+
+        Args:
+            rewards: list of episode rewards
+            values: list of state values from critic
+            last_value: value of last state of episode
+
+        Returns:
+            list of advantages
+        """
+        rews = rewards + [last_value]
+        vals = values + [last_value]
+        # GAE
+        delta = [rews[i] + self.gamma * vals[i + 1] - vals[i] for i in range(len(rews) - 1)]
+        adv = self.discount_rewards(delta, self.gamma * self.lam)
+
+        return adv
+
+    def generate_trajectory_samples(self) -> Tuple[List[torch.Tensor], List[torch.Tensor], List[torch.Tensor]]:
```

Comment:
this method returns tuple of 5 lists instead of 3. id fix the typing accordingly but it exceeds max line length. any suggestions of what can be done?

Reply:
@sid-sundrani You can just add types and push without thinking about the max line length. pre-commit.ci should run a code formatter `yapf` and push the change automatically for you :]